lms/djangoapps/course_api/blocks/transformers/block_completion.py

```diff
@@ -58,14 +69,13 @@ def mark_complete(self, complete_course_blocks, latest_complete_block_key, block
         """
         if block_key in complete_course_blocks:
             block_structure.override_xblock_field(block_key, self.COMPLETE, True)
-            if block_key == latest_complete_block_key:
+            if str(block_key) == str(latest_complete_block_key):
```

Comment:
@Dillon-Dumesnil, @cpennington, I'm not sure whether it's the right solution, but when BlockCompletions are [retrieved](https://github.com/open-craft/edx-platform/blob/747e0ae9d8d38f124a44653d5ba0cbcc902f48ad/lms/djangoapps/course_api/blocks/transformers/block_completion.py#L111-L1170) in the transformer, the block_key that is [stored in its model](https://github.com/edx/completion/blob/master/completion/models.py#L197) as [UsageKeyField](https://github.com/edx/opaque-keys/blob/master/opaque_keys/edx/django/models.py#L200-L205) does not have its attributes - specifically the course run is missing for [old Mongo keys](https://github.com/edx/edx-platform/blob/3030efec78b77bcfcb363b0c3e7fb99dce9df82c/openedx/features/course_experience/tests/views/test_course_outline.py#L529-L531), as described [here](https://github.com/edx/completion/blob/adcaad8c5abd037004c4eda70b647f7a6775615f/completion/models.py#L52-L57). So it looks like some information from block_key are not being saved into the DB at all. However for this use case this workaround seems to be working correctly.

Reply:
My concern would be what the behavior is when an old-mongo key hits this in production. We still have a small number of old-mongo courses live in the wild, and there may be more in the Open edX ecosystem. It may be worth doing the work to re-hydrate the course run info into the old Mongo keys when doing this comparison.

Reply:
FIWW, Old Mongo is basically at end-of-life, and a number of TNL projects have happened where we simply don't support Old Mongo at all. It doesn't break existing content, but new features (like the courseware MFE) are just unavailable for it.

Reply:
As long as nothing existing doesn't break, not supporting old mongo is fine. We _ought_ to be removing old mongo soon, although I can't give a timeline at this moment.